language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/dependencygraph/DependencyGraph.scala

```diff
@@ -7,23 +7,24 @@ import com.daml.lf.codegen.Util
 import com.daml.lf.data.ImmArray.ImmArraySeq
 import com.daml.lf.codegen.lf.DefTemplateWithRecord
 import com.daml.lf.data.Ref.Identifier
-import com.daml.lf.iface.{EnvironmentInterface, InterfaceType}
+import com.daml.lf.iface.{InterfaceType, DefDataType}
+import scalaz.std.either._
 import scalaz.std.list._
 import scalaz.syntax.bifoldable._
+import scalaz.syntax.bifunctor._
 import scalaz.syntax.foldable._
 
 private[codegen] object DependencyGraph {
   def orderedDependencies(
-      library: EnvironmentInterface
-  ): OrderedDependencies[Identifier, TypeDeclOrTemplateWrapper[DefTemplateWithRecord]] = {
-    val decls = library.typeDecls
+      decls: Map[Identifier, InterfaceType]
+  ): OrderedDependencies[Identifier, Either[DefTemplateWithRecord, DefDataType.FWT]] = {
```

Comment:
As it turns something like `DefTemplateWithRecord` exists in `//daml-lf/interface` in the same subclassing hierarchy as `DefDataType`. This means that the `Either` is not really needed but I'm leaving this refactoring for a later time since this would have started to go in the inner workings of the Scala codegen and it felt too much for a single PR.